projects/experimental/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -127,11 +133,13 @@ export class TuiInputPhoneInternational extends TuiControl<string> {
         computed(() => this.computeMask(this.code(), this.metadata())),
     );
 
-    protected readonly masked = computed(
-        () =>
-            maskitoTransform(this.value(), this.mask() || MASKITO_DEFAULT_OPTIONS) ||
-            this.el.value,
-    );
+    protected readonly masked = signal(this.el.value);
+
+    protected valueChangeEffect = effect(() => {
+        this.onChange(this.unmask(this.masked()));
+        // Host binding `host: {'[value]': 'masked()'}` has change detection problem with empty string
+        this.render.setProperty(this.el, 'value', this.masked());
```

Comment:
Since we only need it for value property we can make it a host directive to hide this hideous logic inside one place, if we cannot figure out why this issue is happening.

Reply:
https://github.com/taiga-family/taiga-ui/pull/10220/commits/8d6d6b04591e2cc60d203b11b9d6946c8c274296